cmd/fluent-bit/loki_test.go

```diff
@@ -299,3 +299,47 @@ func Test_labelMapping(t *testing.T) {
 		})
 	}
 }
+
+func Test_AutoKubernetesLabels(t *testing.T) {
+	tests := []struct {
+		name    string
+		records map[interface{}]interface{}
+		want    model.LabelSet
+	}{
+		{
+			"records without labels",
+			map[interface{}]interface{}{
+				"kubernetes": map[interface{}]interface{}{
+					"foo": []byte("buzz"),
+				},
+			},
+			model.LabelSet{
+				"foo": "buzz",
+			},
+		},
+		{
+			"records with labels",
+			map[interface{}]interface{}{
+				"kubernetes": map[string]interface{}{
```

Comment:
Can you add a test without kubernetes key basically an empty map. I have a feeling it will fail.

Reply:
Yes definitely it will fail. We should check if `kubernetes` key is present when `AutoKubernetesLabels` is turned on. Will add that logic

Reply:
you can just `kube,ok := records["kubernetes"]` and skip if the key is missing.

Reply:
With current logic, if autokuberneteslabels is on, then no other labels will be extracted. So, we will have entries with empty labels. This would not be desired behaviour. 
May be we should check `kube,ok := records["kubernetes"]` here https://github.com/grafana/loki/blob/master/cmd/fluent-bit/loki.go#L44 before we call `autokuberneteslabels` . If `!ok` do `lbs = extractLabels(records, l.cfg.labelKeys)` https://github.com/grafana/loki/blob/master/cmd/fluent-bit/loki.go#L48

Reply:
I think the current implementation is fine, but add a test for error case.

Reply:
updated my code. Also, added test case for missing k8s key